packages/frontend/model/validate.ts

```diff
@@ -17,11 +18,17 @@ export const validateAsCAPIType = (data: any): CAPIType => {
 
     if (!isValid) {
         const url = data.webURL || 'unknown url';
+        const errorMessage = `Unable to validate request body for url ${url}.\n${JSON.stringify(
+            validate.errors,
+            null,
+            2,
+        )}`;
 
-        throw new TypeError(
-            `Unable to validate request body for url ${url}.\n
-            ${JSON.stringify(validate.errors, null, 2)}`,
-        );
+        if (process.env.NODE_ENV === 'development') {
```

Comment:
Possibly just !== 'production' ?

Reply:
Yes

Reply:
> I feel like I need convincing on this.
> 
> There are cases where we _can_ recover from something not being quite right in the data model, but there are going to be other cases where something really is fundamentally broken on the data in a way we can't recover, and if we ignore those then we run the risk of not seeing real major production issues.

I just had a similar chat with Nick. It's a fair point, if the data is gravely corrupted then the user experience could be bad. But on the other hand, in the 2 recent cases we've had where this validation function caused 500s, the page could have loaded successfully so we were serving 500s when we didn't need to.

Ideally, this kind of validation should happen further down the stack, closer to the individual components, so that if there's an issue we can render some polite html only for that section of the page. We could still lightly validate the main structure of the CAPI response at this stage, but leave the nitty gritty of checking each element to those components that use that data.

As for the risk that these issues are not seen, I'm using `logger` here which is writing to `/var/log`. If this process were to write to something like an ELK stack or another log service we could track issues better. But yes, it's definitely the case that if we throw 500s then we will respond faster

Reply:
@nicl @MatthewJWalls Lets park this for now and have a chat to figure out what we think is the best way forward

Reply:
Just on the logging - if you use `logger` it does go to ELK, though I never managed to make it pick up exceptions so you have to explicitly log stuff. You can see the logs here: https://logs.gutools.co.uk/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-44m,to:now))&_a=(columns:!(_source),index:a35a6090-59d7-11e8-bbe4-cbb5b151b19c,interval:auto,query:(language:lucene,query:'app:dotcom-rendering'),sort:!('@timestamp',desc))